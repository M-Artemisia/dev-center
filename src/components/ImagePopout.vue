<template>
  <div class="image-container">
    <div class="syn-flex direction-column align-center syn-mv-16">
      <div @click="modalOpen = true" ref="image" class="syn-image-shadowed">
        <slot></slot>
      </div>
      <div v-if="caption" class="syn-overline syn-text-center syn-mt-16">
        {{caption}}
      </div>
    </div>
    <modal v-model="modalOpen" no-padding hide-close-btn large align-center card-view-sm>
      <div @click="modalOpen = false">
        <slot></slot>
      </div>
    </modal>
  </div>
</template>

<script>
// This component handles the caption and popout for an individual image.
// ImagesSection handles the gray bg and formatting multiple images in a row.
import Modal from '../../synapse/components/vue/Modal.vue'

export default {
  name: 'ImpagePopout',
  props: {
    caption: {
      type: String,
      required: false
    },
  },
  components: { Modal },
  data() {
    return {
      modalOpen: false,
    }
  }
}
</script>
