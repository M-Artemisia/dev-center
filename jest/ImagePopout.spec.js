import { mount } from '@vue/test-utils'
import ImagePopout from '@/components/ImagePopout'

describe('ImagePopout', () => {
  test('should fill slot in container and modal with image provided', () => {
    const wrapper = mount(ImagePopout, {
      slots: {
        default: `<img>`
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should display caption if provided', () => {
    const wrapper = mount(ImagePopout, {
      propsData: {
        caption: "Image Caption"
      },
      slots: {
        default: `<img>`
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should open modal when image is clicked', () => {
    const wrapper = mount(ImagePopout, {
      slots: {
        default: `<img>`
      }
    })

    wrapper.find('.syn-image-shadowed').trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should close modal when modal image is clicked', () => {
    const wrapper = mount(ImagePopout, {
      slots: {
        default: `<img>`
      }
    })

    wrapper.find('.syn-image-shadowed').trigger('click')
    wrapper.find('.syn-modal-card div').trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
