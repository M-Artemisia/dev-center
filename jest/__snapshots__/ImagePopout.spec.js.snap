// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePopout should close modal when modal image is clicked 1`] = `
<div class="image-container">
  <div class="syn-flex direction-column align-center syn-mv-16">
    <div class="syn-image-shadowed"><img></div>
    <!---->
  </div>
  <div class="syn-modal" align-center="">
    <div class="syn-modal-card no-padding large card-view-sm fit-content">
      <!---->
      <div><img></div>
    </div>
  </div>
</div>
`;

exports[`ImagePopout should display caption if provided 1`] = `
<div class="image-container">
  <div class="syn-flex direction-column align-center syn-mv-16">
    <div class="syn-image-shadowed"><img></div>
    <div class="syn-overline syn-text-center syn-mt-16">
      Image Caption
    </div>
  </div>
  <div class="syn-modal" align-center="">
    <div class="syn-modal-card no-padding large card-view-sm fit-content">
      <!---->
      <div><img></div>
    </div>
  </div>
</div>
`;

exports[`ImagePopout should fill slot in container and modal with image provided 1`] = `
<div class="image-container">
  <div class="syn-flex direction-column align-center syn-mv-16">
    <div class="syn-image-shadowed"><img></div>
    <!---->
  </div>
  <div class="syn-modal" align-center="">
    <div class="syn-modal-card no-padding large card-view-sm fit-content">
      <!---->
      <div><img></div>
    </div>
  </div>
</div>
`;

exports[`ImagePopout should open modal when image is clicked 1`] = `
<div class="image-container">
  <div class="syn-flex direction-column align-center syn-mv-16">
    <div class="syn-image-shadowed"><img></div>
    <!---->
  </div>
  <div class="syn-modal open" align-center="">
    <div class="syn-modal-card no-padding large card-view-sm fit-content">
      <!---->
      <div><img></div>
    </div>
  </div>
</div>
`;
